Jenkinsfile

```diff
@@ -1,52 +1,59 @@
 elifePipeline {
     node('containers-jenkins-plugin') {
         def commit
-        def grobidTag
-        def fullImageTag
+        def allGrobidTags
 
         stage 'Checkout', {
             checkout scm
             commit = elifeGitRevision()
-            grobidTag = sh(
-                script: 'bash -c "source .env && echo \\$GROBID_TAG"',
+            allGrobidTags = sh(
+                script: 'bash -c "source .env && echo \\$ALL_GROBID_TAGS"',
                 returnStdout: true
-            ).trim()
-            echo "GROBID_TAG: ${grobidTag}"
-            assert grobidTag != ''
-            fullImageTag = "${grobidTag}-${commit}"
-            echo "Full image tag: ${fullImageTag}"
+            ).trim().split(':')
+            echo "allGrobidTags: ${allGrobidTags}"
+            assert allGrobidTags != []
         }
 
         stage 'Build and run tests', {
             try {
-                sh "IMAGE_TAG=${fullImageTag} REVISION=${commit} make ci-build-and-test"
+                parallel(allGrobidTags.inject([:]) { m, grobidTag ->
+                    m["Build and run tests (${grobidTag})"] = {
+                        def fullImageTag = "${grobidTag}-${commit}"
+                        sh "GROBID_TAG=${grobidTag} IMAGE_TAG=${fullImageTag} REVISION=${commit} make ci-build-and-test"
```

Comment:
I wonder if this is free from race conditions. The build phase should be if the image names or tags contain `GROBID_TAG`. Running the tests, to be checked depending on which containers are created and whether they have isolated names and depend on similarly isolated containers.

Reply:
https://github.com/elifesciences/sciencebeam-trainer-grobid/blob/develop/Makefile seems to execute tasks mostly in the container, which helps isolation. The name of that container is generated to be `..._sciencebeam-trainer-grobid_run_1` or similar, I would make this explicit by [passing `--name` to `docker-compose run`](https://github.com/elifesciences/sciencebeam-trainer-grobid/blob/develop/Makefile#L5).

Reply:
It's also failing because of lack of memory (`There is insufficient memory for the Java Runtime Environment to continue.`). I am thinking whether it might be best to just build in parallel but run the test for only one version.

Reply:
Otherwise we will also need to solve the conflict on the memory. In this case it will need access to the network.

Reply:
The `containers` instances are standardized so not easy to expand the memory without incurring more costs in all the other projects.